Source/common/new.santa.proto

```diff
@@ -0,0 +1,283 @@
+  string common_name = 2;
+}
+
+message Execution {
+  enum Decision {
+    DECISION_UNKNOWN = 0;
+    DECISION_ALLOW = 1;
+    DECISION_DENY = 2;
+  }
+
+  enum Reason {
+    REASON_UNKNOWN = 0;
+    REASON_BINARY = 1;
+    REASON_CERT = 2;
+    REASON_COMPILER = 3;
+    REASON_NOT_RUNNING = 4;
+    REASON_PENDING_TRANSITIVE = 5;
+    REASON_SCOPE = 6;
+    REASON_TEAM_ID = 7;
+    REASON_TRANSITIVE = 8;
+    // TODO: Support this
+    REASON_OTHER_ES_CLIENT = 9;
+  }
+
+  enum Mode {
+    MODE_UNKNOWN = 0;
+    MODE_LOCKDOWN = 1;
+    MODE_MONITOR = 2;
+  }
+
+  ProcessInfo instigator = 1;
+  ProcessInfo target = 2;
+
+  File script = 3;
+  File workingDirectory = 4;
+
+  repeated string args = 5;
+  repeated string envs = 6;
+  repeated FileDescriptor fds = 7;
+
+  Decision decision = 8;
+  Reason reason = 9;
+  Mode mode = 10;
+  CertificateInfo certificate_info = 11;
+
+  string explain = 12;
+  string quarantine_url = 13;
+  string original_path = 14;
+}
+
+message Fork {
+  ProcessInfo instigator = 1;
+  ProcessInfo child = 2;
+}
+
+// If ExitType is EXITED, `value` is the exit status
+// If ExitType is SIGNALED or STOPPED, `value` is the signal received
+message Exit {
+  enum ExitType {
+    EXIT_UNKNOWN = 0;
+    EXIT_EXITED = 1;
+    EXIT_SIGNALED = 2;
+    EXIT_STOPPED = 3;
+  }
+
+  ProcessInfo instigator = 1;
+  ExitType type = 2;
+  int32 value = 3;
+}
+
+message Open {
+  ProcessInfo instigator = 1;
+  File target = 2;
+  int32 flags = 3;
+}
+
+message Close {
+  ProcessInfo instigator = 1;
+  File target = 2;
+  bool modified = 3;
+}
+
+// If the destination file already exists, `destination` will be populated
+// Else `directory` and `filename` will be provided.
+message Rename {
+  ProcessInfo instigator = 1;
+  File source = 2;
+  File destination = 3;
+  File destination_directory = 4;
+  string destination_filename = 5;
+}
+
+message Unlink {
+  ProcessInfo instigator = 1;
+  File target = 2;
+}
+
+message Link {
+  ProcessInfo instigator = 1;
+  File source = 2;
+  File target_directory = 3;
+  string target_filename = 4;
+}
+
+message Disk {
+  enum Action {
+    ACTION_UNKNOWN = 0;
+    ACTION_APPEARED = 1;
+    ACTION_DISAPPEARED = 2;
+  }
+
+  Action action = 1;
+  string mount = 2;
+  string volume = 3;
+  string bsd_name = 4;
+  string fs = 5;
+  string model = 6;
+  string serial = 7;
+  string bus = 8;
+  string dmg_path = 9;
+  string appearance = 10;
+}
+
+message Bundle {
+  // This is the hash of the file within the bundle that triggered the event
+  Hash file_hash = 1;
+  // This is the hash of the hashes of all executables in the bundle
+  Hash bundle_hash = 2;
+  string bundle_name = 3;
+  string bundle_id = 4;
+  string bundle_path = 5;
+  string path = 6;
+}
+
+message Allowlist {
+  int32 pid = 1;
+  int32 pidversion = 2;
+  string path = 3;
+  Hash hash = 4;
+}
+
+message SantaMessage {
+  // Unique, per-event generated.
+  // Can allow client to relate events across time
+  string uuid = 1;
```

Comment:
Are we going to generate these or do they come from ES?

Reply:
We would generate these. Can chat if we think it's useful or not. Could be used to tie multiple events together in the future (as a sort of FK). Also for a search index on a backend.